### PR TITLE
[CBRD-22736] Regu variable own implementation file

### PR DESCRIPTION
**Pull request by ribram**
http://jira.cubrid.org/browse/CBRD-22736

Create an implementation file for regular variables where we can add more functionality in the future.

### Changes:

1. Convert regu_variable_node from struct to class and rename regu_var.h to hpp. These two changes touch most of files.
2. Move reguvar tree mapping functions from fetch.c to regu_var.cpp/hpp - names as map_regu. Include TYPE_REGUVAL_LIST/TYPE_REGU_VAR_LIST in mapping. Add map_regu_and_xasl function to also advance into nested XASL nodes.
3. Add clear_xasl function. Not used yet, but it may help in the future (e.g. xts_debug_check).
4. Remove unused arith_list_node.next fields. Affected files/functions:

    - xasl_generation.c / pt_make_regu_arith, pt_make_regu_pred
    - xasl_regu-alloc.cpp / regu_init (arith_list_node &)
    - query_dump.c / qdump_print_arith_expression
    - query_executor.c / qexec_clear_arith_list
    - stream_to_xasl.c / stx_build_arith_type
    - xasl_to_stream.c / xts_process_arith_type

5. Add clear_xasl function to pred_expr; xasl_predicate.cpp was added.

### CODE REVIEW

File: src/query/regu_var.cpp [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    {
+      // apply XASL function
+      xasl_func (*xasl, stop);
+      if (stop)
+	{
+	  return;
+	}
+    }
+
+  // apply regu function to me
```
Comment (razvanradulescu) [inline src/query/regu_var.cpp]: From this to end of function the code is common with 
regu_variable_node::map_regu (const map_regu_func_type &func, bool &stop)

File: src/xasl/xasl_predicate.hpp [hunk 2]
```diff
@@ -152,13 +153,16 @@ namespace cubxasl
...
   struct pred_expr
   {
-    union
-    {
-      pred m_pred;
-      eval_term m_eval_term;
-      pred_expr *m_not_term;
-    } pe;
-    TYPE_PRED_EXPR type;
+    public:
```
Comment (razvanradulescu) [inline src/xasl/xasl_predicate.hpp]: Do you intend to transform to class ?
public doesn't seem to be required.

File: src/query/regu_var.cpp (hunk 1 above)
Comment (ribram) [reply src/query/regu_var.cpp]: fixed

File: src/xasl/xasl_predicate.hpp (hunk 2 above)
Comment (ribram) [reply src/xasl/xasl_predicate.hpp]: fixed

**Pull request merged**